content-src/components/NewTabPage/NewTabPage.js

```diff
@@ -36,8 +36,8 @@ const NewTabPage = React.createClass({
     document.title = "New Tab";
     setFavicon("newtab-icon.svg");
   },
-  componentDidUpdate() {
-    if (this.props.isReady && !this.state.renderedOnce) {
+  componentWillReceiveProps(nextProps) {
```

Comment:
The react eslint rule has some information https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-did-update-set-state.md and the descriptions of nearby lifecycle methods provides some context https://facebook.github.io/react/docs/component-specs.html#updating-componentdidupdate

Main thing is that calling `setState` from `componentDidUpdate` will cause `render` to run again as something in `props` or `state` changed. Calling `setState` from `componentWillReceiveProps` will effectively coalesce the two `render` calls into one as it will appear as if the `props` and `state` changed at the same time.